src/main/java/io/cdap/plugin/http/common/http/OAuthUtil.java

```diff
@@ -70,12 +78,77 @@ public static AccessToken getAccessToken(BaseHttpConfig config) throws IOExcepti
         return OAuthUtil.getAccessTokenByServiceAccount(config);
       case OAUTH2:
         try (CloseableHttpClient client = HttpClients.createDefault()) {
-          return OAuthUtil.getAccessTokenByRefreshToken(client, config);
+          return getAccessToken(client, (BaseHttpSourceConfig) config);
         }
     }
     return null;
   }
 
+  public static AccessToken getAccessToken(CloseableHttpClient httpclient,
+      BaseHttpSourceConfig config) throws IOException {
+    switch (config.getOauth2GrantType()) {
+      case REFRESH_TOKEN:
+        return getAccessTokenByRefreshToken(httpclient, config);
+      case CLIENT_CREDENTIALS:
+        return getAccessTokenByClientCredentials(httpclient, config.getTokenUrl(),
+            config.getClientId(), config.getClientSecret(), config.getScopes(),
+            config.getOauth2ClientAuthentication().getValue());
+      default:
+        throw new IOException("Invalid Grant Type. Cannot retrieve access token.");
+    }
+  }
+
+  private static AccessToken getAccessTokenByClientCredentials(CloseableHttpClient httpclient,
+      String tokenUrl, String clientId, String clientSecret, String scope,
+      String clientAuthentication) throws IOException {
+    URI uri;
+    HttpPost httppost;
+    try {
+      if (Objects.equals(clientAuthentication, OAuthClientAuthentication.BODY.getValue())) {
+        uri = new URIBuilder(tokenUrl).build();
+        List<BasicNameValuePair> nameValuePairs = new ArrayList<>();
+        nameValuePairs.add(new BasicNameValuePair("scope", scope));
+        nameValuePairs.add(
+            new BasicNameValuePair("grant_type", OAuthGrantType.CLIENT_CREDENTIALS.getValue()));
+        nameValuePairs.add(new BasicNameValuePair("client_id", clientId));
+        nameValuePairs.add(new BasicNameValuePair("client_secret", clientSecret));
+        httppost = new HttpPost(uri);
+        httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
+      } else {
+        uri = new URIBuilder(tokenUrl).setParameter("client_id", clientId)
+            .setParameter("client_secret", clientSecret)
+            .setParameter("grant_type", OAuthGrantType.CLIENT_CREDENTIALS.getValue()).build();
+        httppost = new HttpPost(uri);
+      }
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException(
+          "Failed to build access token URI for OAuth2 with grant type = "
+              + OAuthGrantType.CLIENT_CREDENTIALS.getValue(), e);
+    }
+
+    CloseableHttpResponse response = httpclient.execute(httppost);
+    String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
+
+    JsonElement accessTokenElement = JSONUtil.toJsonObject(responseString).get("access_token");
```

Comment:
below code is getting repeated in both refresh token and client credetials method, move it to a separate method